code/modules/bitrunning/virtual_domain/modular_mob_segment.dm

```diff
@@ -3,27 +3,49 @@
 #define SPAWN_UNLIKELY 35
 #define SPAWN_RARE 10
 
+
+/// Handles spawning mobs for this landmark. Sends a signal when done.
+/obj/effect/landmark/bitrunning/mob_segment/proc/spawn_mobs(turf/origin, datum/modular_mob_segment/segment) as /list
+	var/list/mob/living/spawned_mobs = list()
+
+	spawned_mobs += segment.spawn_mobs(origin)
+
+	SEND_SIGNAL(src, COMSIG_BITRUNNING_MOB_SEGMENT_SPAWNED, spawned_mobs)
+
+	var/list/datum/weakref/mob_refs = list()
+	for(var/mob/living/spawned as anything in spawned_mobs)
+		if(QDELETED(spawned))
+			continue
+
+		mob_refs += WEAKREF(spawned)
+
+	return mob_refs
+
+
+/**
+ * A list for mob spawning landmarks to use.
+ */
 /datum/modular_mob_segment
-	/// Spawn no more than this amount
-	var/max = 4
 	/// Set this to false if you want explicitly what's in the list to spawn
 	var/exact = FALSE
 	/// The list of mobs to spawn
-	var/list/mob/living/mobs = list()
-	/// The mobs spawned from this segment
-	var/list/spawned_mob_refs = list()
+	var/list/mobs = list()
+	/// Spawn no more than this amount
+	var/max = 4
 	/// Chance this will spawn (1 - 100)
 	var/probability = SPAWN_LIKELY
 
+
 /// Spawns mobs in a circle around the location
-/datum/modular_mob_segment/proc/spawn_mobs(turf/origin)
+/datum/modular_mob_segment/proc/spawn_mobs(turf/origin) as /list
```

Comment:
Do we need the `as /list` parameter here?

Reply:
it conveys no benefit, no